src/main/java/org/folio/rest/impl/EholdingsProvidersImpl.java

```diff
@@ -73,26 +74,41 @@ public void getEholdingsProviders(String q, String sort, int page, int count, Ma
           .respond200WithApplicationVndApiJson(converter.convert(vendors)))))
       .exceptionally(e -> {
         logger.error(GET_PROVIDERS_ERROR_MESSAGE, e);
-        asyncResultHandler.handle(Future.succeededFuture(GetEholdingsProvidersResponse
-          .status(500)
-          .header("Content-Type", "application/vnd.api+json")
-          .entity(ErrorUtil.createError(e.getCause().getMessage()))
-          .build()));
+        if(e.getCause() instanceof RMAPIServiceException){
+          RMAPIServiceException rmApiException = (RMAPIServiceException)e.getCause();
+          asyncResultHandler.handle(Future.succeededFuture(Response
+            .status(rmApiException.getRMAPICode())
```

Comment:
I do not remember exactly what status codes RM API gives - are those regular HTTP status codes or internal codes of RM API?

Reply:
Based on the above response, not sure if we should use RM API Code here to return status

Reply:
yes, it's http status code